File: src/samples/cpp+py/sample/scripts/reactive_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from common_msgs.msg import Object  

class MinimalPublisher(Node):

    def __init__(self):
        super().__init__('minimal_publisher')
        self.publisher_ = self.create_publisher(Object, 'topic', 10)     # CHANGE
        timer_period = 0.5
        self.timer = self.create_timer(timer_period, self.timer_callback)
        self.i = 0

    def timer_callback(self):
        msg = Object()                                         
        msg.id = self.i                                     
        self.publisher_.publish(msg)
        self.get_logger().info('Publishing: "%d"' % msg.id) 
        self.i += 1


def main(args=None):
    rclpy.init(args=args)

    minimal_publisher = MinimalPublisher()

    rclpy.spin(minimal_publisher)

    minimal_publisher.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: src/lidar_object_detection/src/reactive_node.cpp
// #include <chrono>
// #include <memory>

// #include "rclcpp/rclcpp.hpp"
// #include "common_msgs/msg/object.hpp"

// using namespace std::chrono_literals;

// class MinimalPublisher : public rclcpp::Node
// {

// private:
//     void timer_callback()
//     {
//         auto message = common_msgs::msg::Object();                   // CHANGE
//         message.id = this->count_++;                                     // CHANGE
//         RCLCPP_INFO(this->get_logger(), "Publishing: '%d'", message.id); // CHANGE
//         publisher_->publish(message);
//     }
//     rclcpp::TimerBase::SharedPtr timer_;
//     rclcpp::Publisher<common_msgs::msg::Object>::SharedPtr publisher_; // CHANGE
//     size_t count_;

// public:
//     MinimalPublisher()
//         : Node("minimal_publisher"), count_(0)
//     {
//         publisher_ = this->create_publisher<common_msgs::msg::Object>("topic", 10); // CHANGE
//         timer_ = this->create_wall_timer(
//             500ms, std::bind(&MinimalPublisher::timer_callback, this));
//     }

// };

// int main(int argc, char *argv[])
// {
//     rclcpp::init(argc, argv);
//     rclcpp::spin(std::make_shared<MinimalPublisher>());
//     rclcpp::shutdown();
//     return 0;
// }